### PR TITLE
Bump `ktlint` to 0.46.1

### PR DESCRIPTION
**Pull request by mateuszkwiecinski**
\+ Replace removed `Ktlint.Params` with the new `Ktlint.ExperimentalParams`

The same as https://github.com/jeremymailen/kotlinter-gradle/pull/257, but also adjusts the plugin to the recent API changes 👀

### CODE REVIEW

File: src/main/kotlin/org/jmailen/gradle/kotlinter/support/KtLintEditoConfigOverrides.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.pinterest.ktlint.core.api.EditorConfigOverride.Companion.plus
+
+internal fun editorConfigOverride(ktLintParams: KtLintParams): EditorConfigOverride {
+    val rules = ktLintParams.disabledRules
+
+    return if (rules.isEmpty()) {
+        emptyEditorConfigOverride
+    } else {
+        EditorConfigOverride.from(disabledRulesProperty to rules.joinToString(separator = ","))
+    }
```
Comment (mateuszkwiecinski) [inline src/main/kotlin/org/jmailen/gradle/kotlinter/support/KtLintEditoConfigOverrides.kt:15]: I followed the migration guide from [the release notes](https://github.com/pinterest/ktlint/releases/tag/0.46.0)

File: src/test/kotlin/org/jmailen/gradle/kotlinter/support/RuleSetsTest.kt [hunk 2]
```diff
@@ -69,5 +69,5 @@ class TestRule(id: String) : Rule(id) {
         node: ASTNode,
         autoCorrect: Boolean,
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
-    ) {}
+    ) = Unit
```
Comment (mateuszkwiecinski) [inline src/test/kotlin/org/jmailen/gradle/kotlinter/support/RuleSetsTest.kt:72]: IJ kept re-formatting the file for me, causing an unnecessary diff :/

**Review by jeremymailen — APPROVED** [review]
Looks good, feel free to merge.

Looks like once we bump the version and release, we'll need to increase the min kotlin version from 1.6.21 -> 1.7.0.

I also noticed this new version of ktlint is pretty aggressive about formatting function parameters to be on separate lines. A lot of changes for some of my projects, but this isn't a plugin consideration, I'm sure the change was well vetted.
